includes/admin/home/tasks/task/class-sensei-home-task-create-first-course.php

```diff
@@ -0,0 +1,88 @@
+<?php
+/**
+ * File containing the Sensei_Home_Task_Create_First_Course class.
+ *
+ * @package sensei-lms
+ * @since $$next-version$$
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+/**
+ * Sensei_Home_Task_Create_First_Course class.
+ *
+ * @since $$next-version$$
+ */
+class Sensei_Home_Task_Create_First_Course implements Sensei_Home_Task {
+	/**
+	 * The ID for the task.
+	 *
+	 * @return string
+	 */
+	public static function get_id(): string {
+		return 'create-first-course';
+	}
+
+	/**
+	 * Number used to sort in frontend.
+	 *
+	 * @return int
+	 */
+	public function get_priority(): int {
+		return 200;
+	}
+
+	/**
+	 * Task title.
+	 *
+	 * @return string
+	 */
+	public function get_title(): string {
+		return __( 'Create your first Course', 'sensei-lms' );
+	}
+
+	/**
+	 * Task url.
+	 *
+	 * @return string
+	 */
+	public function get_url(): ?string {
+		return admin_url( 'post-new.php?post_type=course' );
+	}
+
+	/**
+	 * Task image.
+	 *
+	 * @return string|null
+	 */
+	public function get_image(): ?string {
+		return null;
+	}
+
+	/**
+	 * Whether the task is completed or not.
+	 *
+	 * @return bool
+	 */
+	public function is_completed(): bool {
+		global $wpdb;
+		$cache_key   = 'home/tasks/create-first-course';
+		$cache_group = 'sensei/temporary';
+		$result      = wp_cache_get( $cache_key, $cache_group );
+		if ( false === $result ) {
+			$prefix = $wpdb->esc_like( Sensei_Data_Port_Manager::SAMPLE_COURSE_SLUG );
+			// phpcs:ignore WordPress.DB.DirectDatabaseQuery.DirectQuery -- Safe-ish and rare query.
+			$result = $wpdb->get_var( $wpdb->prepare( "SELECT COUNT(*) FROM {$wpdb->posts} WHERE post_type='course' AND post_name NOT LIKE %s", "{$prefix}%" ) );
+			if ( null === $result ) {
```

Comment:
In my tests, this returns `0` when no posts, so we would also be caching `0` "forever" on object caching sites.

Reply:
Hey there, technically, this shouldn't cache forever, as I'm using `sensei/temporary` as cache group, and that cache group is added as non persistent (using this [WP function](https://developer.wordpress.org/reference/functions/wp_cache_add_non_persistent_groups/)) here: https://github.com/Automattic/sensei/blob/c50c2339b57308fa6f6e69a0892eedc361bdd4f1/includes/class-sensei.php#L350

Reply:
Oh interesting! I think we'll need to still be careful with it. Although this screen might not matter, that function still doesn't do anything on WPCOM for some reason (I'll send link). It is implemented on Atomic, though.

Reply:
Fair enough. I created #5935 to handle this. THanks!